### ISSUE TITLE
Inference denoise example

### ISSUE THREAD

**Issue by betegon** [anicolson/DeepXi#28]
Hi @anicolson ,

I will want to know if you could write in a more descriptive way,  how to do inference with pre-trained weights provided (`3f`). I wrote about how to achieve an inference in an already closed issue  #19 , but there is still gaps to fill.

This will allow people to give it a quick try at the repo and keep digging in. 

**What about the characteristics of the input audio to perform inference?**
**1.** Should it have a determined `sample rate`,  _e.g._ `16000 Hz` ?
**2.** Should it have a determined `bit rate`,  _e.g._ `256 kbps` ?

**Is `mmse-lsa` the best gain function to use?**
As per the result's table in the `README` it looks like, but it is also using `ResNet 3e`, what about `3f`?

**What should be the expected result of the inference if performed correctly? Could you provide a test audio and a result audio using the weights of `3f` 175 epochs + `mmse-lsa`?**
This way, we could know if we have the things set up correctly.  Personally, I have try to infer some audios and I am not sure if it is the expected result or know, so this will be a good way to check that.
 
Also, I would like to offer my help (if you allow people to contribute) by adding utils script to allow users to input any kind of audio and convert into the optimal one for the network once these questions are answered.

Thank you so much for you hard work and effort,

kind regards.

**Comment by anicolson** [comment]
Deep Xi in its current configuration operates on mono/single-channel audio. Single-channel is commonly used in speech enhancement due to most cell phone microphone configurations (single microphone).

Version 3f is trained to operate on a sampling frequency of 16 kHz, which is currently the standard sampling frequency used in the speech enhancement community (previously the standard sampling frequency was 8 kHz). Deep Xi can be trained using a higher sampling frequency, but this would be unnecessary as human speech rarely exceeds 8 kHz (the Nyquist frequency of a sampling frequency of 16 kHz). 

Bit rate is different for each audio codec. As Deep Xi uses PySoundFile, a variety of audio codecs can be used (e.g. .wav, .mp3, .flac, etc). PySoundFile then converts the coded audio to 16-bit PCM (int16). Deep Xi then converts this to float32 and then normalises to [-1.0,1.0]. In short, the bit rate does not affect the performance of Deep Xi. A lossy codec may very, very slightly affect the original audio, however. 

Hope this helps.

**Comment by betegon** [comment]
Sorry for the late response.

Thank you for the insight, but you only answered the first question, so I imagine there won't be any test audio + its results.

One way or the other, thank you for putting this out.

King regards.

**Comment by anicolson** [comment]
After I finish updating Deep Xi, i'll find time to include more results and files. If you want to know the performance of different MMSE approaches/gain functions, please read the original paper:

https://github.com/anicolson/DeepXi/blob/master/1-s2.0-S0167639318304308-main.pdf
________________________________
